src/MainWindow.cpp

```diff
@@ -3165,145 +3165,159 @@ int CMainWindow::OpenFile(const std::wstring& file, unsigned int openFlags)
+            id = activetabid;
+            m_TabBar.SetCurrentTitle(sTitle.c_str());
+        }
+        doc.SetLanguage(CLexStyles::Instance().GetLanguageForDocument(doc, m_scratchEditor));
 
-            // only activate the new doc tab if the main window is enabled:
-            // if it's disabled, a modal dialog is shown
-            // (e.g., the handle-outside-modifications confirmation dialog)
-            // and we then must not change the active tab.
-            if (IsWindowEnabled(*this))
+        if (CPathUtils::PathCompare(filepath, m_tabmovepath) == 0)
+        {
+            filepath = m_tabmovesavepath;
+        }
+        // check if the file is special, i.e. if we need to do something
+        // when the file is saved
+        auto sExt = CStringUtils::to_lower(CPathUtils::GetFileExtension(doc.m_path));
+        if (sExt == L"bpj" || sExt == L"bpv")
+        {
+            doc.m_saveCallback = [this]() { CCommandHandler::Instance().InsertPlugins(this); };
+        }
+        else if (sExt == L"ini")
+        {
+            if (doc.m_path == CIniSettings::Instance().GetIniPath())
             {
-                bool bResize = m_fileTree.GetPath().empty() && !doc.m_path.empty();
-                if (bActivate)
-                {
-                    m_TabBar.ActivateAt(index);
-                }
-                else
-                    // SCI_SETDOCPOINTER is necessary so the reference count of the document
-                    // is decreased and the memory can be released.
-                    m_editor.Call(SCI_SETDOCPOINTER, 0, doc.m_document);
-                if (bResize)
-                    ResizeChildWindows();
+                doc.m_saveCallback = []() { CIniSettings::Instance().Reload(); };
             }
-            else
-                m_editor.Call(SCI_SETDOCPOINTER, 0, doc.m_document);
+        }
+        else if (sExt == L"bplex")
+        {
+            doc.m_saveCallback = []() { CLexStyles::Instance().Reload(); };
+        }
 
-            CEditorConfigHandler::Instance().ApplySettingsForPath(doc.m_path, &m_editor, doc, false);
-            InvalidateRect(m_TabBar, nullptr, FALSE);
+        m_DocManager.AddDocumentAtEnd(doc, id);
+        doc = m_DocManager.GetDocumentFromID(id);
 
-            if (bAddToMRU)
-                SHAddToRecentDocs(SHARD_PATHW, filepath.c_str());
-            CCommandHandler::Instance().OnDocumentOpen(id);
-            if (m_fileTree.GetPath().empty())
-            {
-                m_fileTree.SetPath(CPathUtils::GetParentDirectory(filepath), false);
+        // only activate the new doc tab if the main window is enabled:
+        // if it's disabled, a modal dialog is shown
+        // (e.g., the handle-outside-modifications confirmation dialog)
+        // and we then must not change the active tab.
+        // SCI_SETDOCPOINTER is necessary so the reference count of the document
+        // is decreased and the memory can be released.
+        if (IsWindowEnabled(*this))
```

Comment:
the comment here clearly states that SCI_SETDOCPOINTER is necessary, but then you removed the call?

Reply:
I removed the SCI_SETDOCPOINTER call, because it was never ever called, it was only called if bActivate was false and it was always true.